packages/loader/core-interfaces/src/encoder.ts

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { IFluidHandle } from "./handles";
+
+export const IFluidHandleEncoder: keyof IProvideFluidHandleEncoder = "IFluidHandleEncoder";
+
+export interface IProvideFluidHandleEncoder {
+    readonly IFluidHandleEncoder: IFluidHandleEncoder;
+}
+
+/**
+ * Encoder used to encode/decode IFluidHandles to/from their jsonable form.
+ *
+ * Typical usage includes:
+ *  - Preparing ops for serialization prior to calling `submitLocalMessage`.
+ *  - Retrieving handles from ops inside `processCore`.
+ *
+ * Also used internally by IFluidSerializer stringify()/parse() to preserve handle
+ * references within summary nodes.
+ */
+export interface IFluidHandleEncoder extends IProvideFluidHandleEncoder {
```

Comment:
do we think this a common enough case, and a pattern we want to encourage such that an interface is necessary?

The other options is just to expose functions like `encode(serializer:IFluidSerializer, value: any, bind: IFluidHandle): any;`

Reply:
I agree. We should probably start with them as utility functions and if need be, we can put it on an interface.
Also, I was thinking, if we can use these in FluidSerializer's `stringify` and `parse`, they become much efficient and then there is less need for this. If someone still wanted to use them, they could use the utility functions.

Reply:
Most DDSes need to call encode/decode.  This is because op submission/processing doesn't go through the Fluid serializer.  Ops are wrapped in message envelopes, etc. at lower layers.  As a result, the final JSON.stringify(..) happens too deep in the stack to know the handle serialization context.

Unfortunately, `decode()` needs a pair of IFluidHandleContexts to decode the handle, making decode not a good candidate for a static helper function.

I can put encode/decode back on IFluidSerializable if that's preferred?